sdk/keyvault/Azure.Security.KeyVault.Secrets/tests/ChallengeBasedAuthenticationPolicyTests.cs

```diff
@@ -343,6 +343,77 @@ public void ThrowsWithTwoConsecutiveCaeChallenges()
             }
         }
 
+        [Test]
+        [TestCase("standard", """Bearer realm="", authorization_uri="https://login.microsoftonline.com/common/oauth2/authorize", error="insufficient_claims", claims="eyJhY2Nlc3NfdG9rZW4iOnsibmJmIjp7ImVzc2VudGlhbCI6dHJ1ZSwidmFsdWUiOiIxNzI2MDc3NTk1In0sInhtc19jYWVlcnJvciI6eyJ2YWx1ZSI6IjEwMDEyIn19fQ==" """, """{"access_token":{"nbf":{"essential":true,"value":"1726077595"},"xms_caeerror":{"value":"10012"}}}""")]
+        public async Task VerifyClaimsInToken(string challenge, string expectedClaims)
+        {
+            string claims = null;
+            int callCount = 0;
+
+            MockTransport keyVaultTransport = new(new[]
+            {
+                // Initial challlenge
+                new MockResponse(401)
+                    .WithHeader("WWW-Authenticate", @"Bearer authorization=""https://login.windows.net/de763a21-49f7-4b08-a8e1-52c8fbc103b4"", resource=""https://vault.azure.net"""),
+
+                // CAE Challenge
+                new MockResponse(401)
+                    .WithHeader("WWW-Authenticate", challenge),
+
+                new MockResponse(200)
+                {
+                    ContentStream = new KeyVaultSecret("test-secret", "secret-value").ToStream(),
+                },
+            });
+
+            var credential = new TokenCredentialStub((r, c) =>
+            {
+                claims = r.Claims;
```

Comment:
The policy should set `Claims` only for the first call after a CAE challenge. This test checks only the last value the policy passed. That means it wouldn't fail if the policy passed nonempty values in previous calls, which would be a bug. So, I suggest making this more precise by considering `callCount`. You expect exactly 2 calls, with no claims in the first one, and the expected claims in the second.